django_components/template_loader.py

```diff
@@ -2,34 +2,84 @@
 Template loader that loads templates from each Django app's "components" directory.
 """
 
+import os
 from pathlib import Path
+from typing import Set
 
+from django.apps import apps
 from django.conf import settings
 from django.template.loaders.filesystem import Loader as FilesystemLoader
-from django.template.utils import get_app_template_dirs
+
+from django_components.logger import logger
+
+
+# Same as `Path.is_relative_to`, defined as standalone function because `Path.is_relative_to`
+# is marked for deprecation.
+def path_is_relative_to(child_path: str, parent_path: str):
+    # If the relative path doesn't start with `..`, then child is descendant of parent
+    # See https://stackoverflow.com/a/7288073/9788634
+    rel_path = os.path.relpath(child_path, parent_path)
+    return not rel_path.startswith("..")
 
 
 class Loader(FilesystemLoader):
     def get_dirs(self):
```

Comment:
!IMPORTANT - PLEASE REVIEW!

There are couple of points related to `Loader.get_dirs` that need to be discussed:

1. I tried to document the flow a bit. However, I'm not sure if I got things right around `settings.SETTINGS_MODULE`, so please give that a look

2. I don't feel like I have a full grasp of the relationship between `STATICFILES_DIRS`, `TEMPLATES.DIRS`, and the `component_dir` variable:

 	Initially, I thought components are configured via `TEMPLATES.DIRS`, because in the project where I introduced django components, the `STATICFILES_DIRS` and `TEMPLATES.DIRS` were colocated.

 	When I started looking into this codebase, I saw that they are configured by `STATICFILES_DIRS`.

 	The `sampleproject` demo confirms that components are configured by `STATICFILES_DIRS`.

 	However, for some reason I had to add [the components dir to `TEMPLATES.DIR` in tests](https://github.com/EmilStenstrom/django-components/pull/395/files#diff-d75294aaaae87b0a95ac59c71cd89d1d288e3670cabb568a294ed2748def5e83R10). And I don't understand why. So this confuses me.

3. Continuing with the changes, I made a change so that the `component_dir` is not hardcoded, but rather used only as a default if `STATICFILES_DIRS` are not set. The reason for this is that, if we go back to point 2), then the components' location is derived from 2-3 variables.

	However, in README the instructions say to use `STATICFILES_DIRS` to configure the components dir, and the example uses a path which already includes the `".../components"` part:

	```python
	STATICFILES_DIRS = [
	    ...,
	    os.path.join(BASE_DIR, "components"),
	]
	```

	So docs make it look like it's define via `STATICFILES_DIRS`, but there are some extra vars behind the scenes.
	So IMO doing away with the hardcoded variable `component_dir` should simplify things, and make the behaviour more likely to be in line with what the docs say.

	So... That's at least the theory. As I said in point 2), don't know this codebase good enough yet to say.

	However, the tests are passing. And the `sampleproject` demo works. So it seems like things are working correctly.

	HOWEVER, this could possibly break for those who defined `STATICFILES_DIRS` but didn't include the components dir (e.g. `"{BASE_DIR}/components"`) among the entries. But then again, docs say that user SHOULD include it.